### PR TITLE
Fix bls to execution change gossip tests

### PR DESCRIPTION
**Pull request by mehdi-aouadi**
Fix the BLS gossip tests around the Capella activation boundary:
Previously, the tests used the Capella spec to create the state while overriding only `CAPELLA_FORK_EPOCH` to `1`. For the mainnet preset, this generated an "incorrect" fork schedule such as:
```yaml
ALTAIR_FORK_EPOCH: 74240
BELLATRIX_FORK_EPOCH: 144896
CAPELLA_FORK_EPOCH: 1
```
This describes Capella activating before Altair and Bellatrix, while the serialized state already uses the Capella SSZ structure.
Teku derives the active milestone and state schema from the fork schedule so the fixture could not be interpreted correctly.
With this change, the fork schedule is now modeled correctly:
```yaml
ALTAIR_FORK_EPOCH: 0
BELLATRIX_FORK_EPOCH: 0
CAPELLA_FORK_EPOCH: 1
```

### CODE REVIEW

File: tests/core/pyspec/eth_consensus_specs/test/capella/networking/test_gossip_bls_to_execution_change.py [hunk 1]
```diff
@@ -68,28 +72,33 @@ def test_gossip_bls_to_execution_change__valid(spec, state):
...
     yield "topic", "meta", "bls_to_execution_change"
     yield "state", state
 
     store, signed_anchor = get_store_from_state(spec, state)
     yield get_filename(signed_anchor), signed_anchor
     yield "blocks", "meta", [{"block": get_filename(signed_anchor)}]
 
-    seen = get_seen(spec)
-    signed_bls_to_execution_change = get_signed_bls_to_execution_change(spec, state)
+    seen = get_seen(post_spec)
```
Comment (jihoonsong) [inline tests/core/pyspec/eth_consensus_specs/test/capella/networking/test_gossip_bls_to_execution_change.py]: I think swapping [these two checks](https://github.com/ethereum/consensus-specs/blob/53dc279a4910331aa6a3453633a9546286d7a5c3/specs/capella/p2p-interface.md?plain=1#L218-L224) would make it unnecessary to use `post_spec` here. Same with L143.

File: tests/core/pyspec/eth_consensus_specs/test/capella/networking/test_gossip_bls_to_execution_change.py [hunk 2]
```diff
@@ -158,30 +172,35 @@ def test_gossip_bls_to_execution_change__ignore_before_clock_disparity(spec, sta
...
 
     store, signed_anchor = get_store_from_state(spec, state)
     yield get_filename(signed_anchor), signed_anchor
     yield "blocks", "meta", [{"block": get_filename(signed_anchor)}]
 
-    seen = get_seen(spec)
-    signed_bls_to_execution_change = get_signed_bls_to_execution_change(spec, state)
-    capella_fork_time_ms = get_capella_fork_time_ms(spec, store)
-    current_time_ms = capella_fork_time_ms - spec.config.MAXIMUM_GOSSIP_CLOCK_DISPARITY
+    seen = get_seen(post_spec)
```
Comment (jihoonsong) [inline tests/core/pyspec/eth_consensus_specs/test/capella/networking/test_gossip_bls_to_execution_change.py:194]: It looks odd to use post-fork's information just for gossip validation.

**Review by jihoonsong — COMMENTED** [review]
I'm not entirely sure about using next fork's structure before the fork transition. Maybe testing this passed-fork edge case is not worth the effort? cc @jtraglia

File: tests/core/pyspec/eth_consensus_specs/test/capella/networking/test_gossip_bls_to_execution_change.py (hunk 1 above)
Comment (mehdi-aouadi) [reply tests/core/pyspec/eth_consensus_specs/test/capella/networking/test_gossip_bls_to_execution_change.py]: addressed, let me know if that's what you had in mind

File: tests/core/pyspec/eth_consensus_specs/test/capella/networking/test_gossip_bls_to_execution_change.py (hunk 2 above)
Comment (mehdi-aouadi) [reply tests/core/pyspec/eth_consensus_specs/test/capella/networking/test_gossip_bls_to_execution_change.py:194]: As discussed, this test requires mixing a Bellatrix state with Capella gossip types for a narrow edge case. I'd suggest removing it to avoid having to do that

File: tests/core/pyspec/eth_consensus_specs/test/capella/networking/test_gossip_bls_to_execution_change.py (hunk 1 above)
Comment (jihoonsong) [reply tests/core/pyspec/eth_consensus_specs/test/capella/networking/test_gossip_bls_to_execution_change.py]: Thanks! Do we fail the test if we don't use post_spec?

Comment (mehdi-aouadi) [reply tests/core/pyspec/eth_consensus_specs/test/capella/networking/test_gossip_bls_to_execution_change.py]: IMU `post_spec` is still needed to construct the message and call the Capella validation function

File: tests/core/pyspec/eth_consensus_specs/test/capella/networking/test_gossip_bls_to_execution_change.py (hunk 2 above)
Comment (mehdi-aouadi) [reply tests/core/pyspec/eth_consensus_specs/test/capella/networking/test_gossip_bls_to_execution_change.py:194]: @jtraglia what do you think about removing this test?

Comment (jtraglia) [reply tests/core/pyspec/eth_consensus_specs/test/capella/networking/test_gossip_bls_to_execution_change.py:194]: Hmm wouldn't this critique apply to all three of these tests? What about this particular test is different? After looking at it, I don't think I want to remove this test case.

Comment (jtraglia) [reply tests/core/pyspec/eth_consensus_specs/test/capella/networking/test_gossip_bls_to_execution_change.py:194]: Maybe we should just remove all three of these tests? And revert the spec change?

**Comment by jtraglia** [comment]
Hey @mehdi-aouadi I merged an alternative to this which simply removes these three test cases.
